### PR TITLE
feat: configure default enabled state

### LINKED ISSUE #16: Allow users to configure default enabled state
Allow users to configure the default enabled state when opening and viewing a markdown file.

https://github.com/MeanderingProgrammer/markdown.nvim/blob/467c13523153f9b918c86037d0b5f2a37094cb88/lua/render-markdown/init.lua#L132

### PR DESCRIPTION
**Pull request by cleong14**
Fixes #16

### CODE REVIEW

File: lua/render-markdown/init.lua [hunk 1]
```diff
@@ -128,8 +129,9 @@ function M.setup(opts)
             latex = '@markup.math',
             quote = '@markup.quote',
         },
+        enabled = false,
```
Comment (MeanderingProgrammer) [inline lua/render-markdown/init.lua:132]: The default value should be `true`.

The plugin should continue to function as it currently does by default and allow users to opt in to the new feature.

File: lua/render-markdown/init.lua [hunk 2]
```diff
@@ -128,8 +129,9 @@ function M.setup(opts)
...
     }
-    state.enabled = true
+    state.enabled = opts and opts.enabled or default_config.enabled
```
Comment (MeanderingProgrammer) [inline lua/render-markdown/init.lua:134]: I don't like the idea of de-syncing the `UserConfig` and overall `Config`.

Let's call this new field something like `start_enabled` or `default_enabled`.

In `state.lua` add the mandatory parameter:

```lua
---@class Config
---@field public start_enabled boolean
```

In `init.lua` add the optional parameter:

```lua
---@class UserConfig
---@field public start_enabled? boolean
```

Set the default value:

```lua
---@type Config
local default_config = {
    start_enabled = true,
```

Update how `state.enabled` is initialized:

```lua
state.config = vim.tbl_deep_extend('force', default_config, opts or {})
state.enabled = state.config.start_enabled
```

**Review by MeanderingProgrammer — CHANGES_REQUESTED** [review]

File: lua/render-markdown/init.lua (hunk 2 above)
Comment (cleong14) [reply lua/render-markdown/init.lua:134]: Ah, I see now. I think I understand why I couldn't get my changes to work at first now. I only made changes to `init.lua` and I missed `state.lua`.

**Comment by cleong14** [comment]
@MeanderingProgrammer changes should be good to go!

File: lua/render-markdown/init.lua (hunk 2 above)
Comment (MeanderingProgrammer) [reply lua/render-markdown/init.lua]: The change to `state.lua` as far as adding the field is just for type annotations, it shouldn't have any functional impacts.

Setting `state.enabled` in `init.lua` is the bit that enables the functionality. Unsure why this didn't seem to work locally.

**Review by MeanderingProgrammer — APPROVED** [review]
LGTM, thanks for making the updates!

**Pull request merged**